ui/src/app/cohort-search/cohort-search/cohort-search.component.ts

```diff
@@ -48,16 +48,19 @@ export class CohortSearchComponent implements OnInit, OnDestroy {
       const cohortId = params.cohortId;
       if (cohortId) {
         this.loading = true;
-        cohortsApi().getCohort(workspace.namespace, workspace.id, cohortId)
-          .then(cohort => {
-            this.loading = false;
-            this.cohort = cohort;
-            currentCohortStore.next(cohort);
-            if (cohort.criteria) {
-              initExisting.next(true);
-              searchRequestStore.next(parseCohortDefinition(cohort.criteria));
-            }
-          });
+        Promise.all([
+          workspacesApi().updateRecentWorkspaces(workspace.namespace, workspace.id),
```

Comment:
If this errors, I'm not sure we want to count all API endpoints as having failed here. This will mean that it rejects if any of the promises fail. I'm not sure we want to do that?

Reply:
Do you know of another way to force all the promises to complete before executing the code in the `.then`?

Reply:
I think the point (which I agree with), is that the result of `updateRecentWorkspaces()` is irrelevant for rendering of the page. It shouldn't be on the critical path at all, I don't think there's anything we need to do in response to that call completing, so I would make the call independently and keep `getCohort()` as it was.